upstairs/src/lib.rs

```diff
@@ -3554,12 +3557,17 @@ impl Downstairs {
                     self.ds_skipped_jobs[cid as usize].insert(io.ds_id);
                 }
                 DsState::LiveRepair => {
-                    let my_limit = self.extent_limit[cid as usize];
+                    // Pick the latest repair limit that's relevant for this
+                    // extent.
+                    let my_limit =
```

Comment:
Yeah, this is cheaper and simpler that the previous code, which checked every single read/write against `repair_job_ids`.  Since `repair_job_ids` is a (sorted) `BTreeMap`, we can just look at the last value to find the farthest extent.